testing/platform/fpga_hw.cpp

```diff
@@ -241,6 +244,53 @@ static T parse_file(const std::string &path) {
   return value;
 }
 
+static std::string make_path(int seg, int bus, int dev, int func){
+    std::stringstream num;
+    num << std::setw(2) << std::hex << bus; 
+    std::string b (num.str());
+    num.clear();
+    num.str(std::string());
+
+    num << std::setw(4) << std::setfill('0') << seg;
+    std::string s (num.str());
+
+    num.clear();
+    num.str(std::string());
+    num << std::setw(2) << std::setfill('0') << dev;
+    std::string d (num.str());
+
+    std::string device_string = s + ":" + b + ":" + d + "." + std::to_string(func);
+    return device_string;
+}
+
+static uint16_t read_socket_id(const std::string devices) {
+  std::string glob_path = PCI_DEVICES + "/" + devices + "/fpga/intel-fpga-dev.*/intel-fpga-fme.*/socket_id";
+  std::string socket_path;
+
+  glob_t glob_buf;
+  glob_buf.gl_pathc = 0;
+  glob_buf.gl_pathv = NULL;
+  int globres = glob(glob_path.c_str(), 0, NULL, &glob_buf);
```

Comment:
Check valgrind on this.  I recall that it's possible to have a negative return value, but for glob to actually allocate data in glob_buf.